### ISSUE TITLE
[JIT] Slice with optional not supported

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#31191]
## 🐛 Tensor Slice with optional 
```
def slice_with_optional(val, start: int, end: Optional[int] = None):
    return val[start:end]

jit.script(slice_with_optional)
```

> Arguments for call are not valid.
The following operator variants are available: 
  aten::slice(Tensor(a) self, int dim=0, int start=0, int end=9223372036854775807, int step=1) -> (Tensor(a)):
  Expected a value of type 'int' for argument 'end' but instead found type 'Optional[int]'.
 ...

Originally reported in https://github.com/pytorch/pytorch/issues/24256

Related issue https://github.com/pytorch/pytorch/issues/27543

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
